docs/how-to/rocm-for-ai/train-a-model.rst

```diff
@@ -2,139 +2,503 @@
+| PyTorch                  | 2.4.0                          |
++--------------------------+--------------------------------+
+| PyTorch Lightning        | 2.4.0                          |
++--------------------------+--------------------------------+
+| Megatron Core            | 0.9.0                          |
++--------------------------+--------------------------------+
+| Transformer Engine       | 1.5.0                          |
++--------------------------+--------------------------------+
+| Flash Attention          | v2.6                           |
++--------------------------+--------------------------------+
+| Transformers             | 4.44.0                         |
++--------------------------+--------------------------------+
 
-PyTorch distributed
--------------------
+Supported features and models
+=============================
+
+Megatron-LM provides the following key features to train large language models efficiently:
 
-As of PyTorch 1.6.0, features in ``torch.distributed`` are categorized into three main components:
+- Transformer Engine (TE)
 
-- `Distributed data-parallel training
-  <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`_ (DDP)
+- APEX
 
-- `RPC-Based distributed training <https://pytorch.org/docs/stable/rpc.html>`_ (RPC)
+- GEMM tuning
 
-- `Collective communication <https://pytorch.org/docs/stable/distributed.html>`_
+- Torch.compile
 
-In this guide, the focus is on the distributed data-parallelism strategy as it’s the most popular. To get started with DDP,
-let’s first understand how to coordinate the model and its training data across multiple accelerators or GPUs.
+- 3D parallelism: TP + SP + CP
 
-The DDP workflow on multiple accelerators or GPUs is as follows:
+- Distributed optimizer
 
-#. Split the current global training batch into small local batches on each GPU. For instance, if you have 8 GPUs and
-   the global batch is set at 32 samples, each of the 8 GPUs will have a local batch size of 4 samples.
+- Flash Attention (FA) 2
 
-#. Copy the model to every device so each device can process its local batches independently.
+- Fused kernels
 
-#. Run a forward pass, then a backward pass, and output the gradient of the weights with respect to the loss of the
-   model for that local batch. This happens in parallel on multiple devices.
+- Pre-training
 
-#. Synchronize the local gradients computed by each device and combine them to update the model weights. The updated
-   weights are then redistributed to each device.
+.. _amd-megatron-lm-model-support:
 
-In DDP training, each process or worker owns a replica of the model and processes a batch of data, then the reducer uses
-``allreduce`` to sum up gradients over different workers.
+The following models are pre-optimized for performance on the AMD Instinct MI300X accelerator.
 
-See the following developer blogs for more in-depth explanations and examples.
+* Llama 2 7B
```

Comment:
Perhaps the Llama should be using Meta on the first occurrence. 
https://ai.meta.com/blog/meta-llama-3/

Reply:
Done @ first occurrence